sympy/solvers/solvers.py

```diff
@@ -1253,6 +1253,7 @@ def _solve(f, *symbols, **flags):
     If no method is implemented to solve the equation, a NotImplementedError
     will be raised. In the case that conversion of an expression to a Poly
     gives None a ValueError will be raised."""
+    f = simplify(f)
```

Comment:
Ok, I am trying to get to the bottom of the problem:

The _solve() method fails for the equation
eq = -a*v - c + (a*b*v - a*v + a – 1)/(b-1)
when it calls for the method solve_linear()   (on the else statement, line 1395, solvers module)

>>> solve_linear(eq,symbols =[a], exclude=[b,c,v])
(0, 1)

To quote from  the documentation string for  the solve_linear() method, on what returning (0,1) means:
<<(0, 1) meaning that ``f`` is independent of the symbols in `symbols`` that aren't in ``exclude``>>.
Obviously eq is not independent of a.

Another example where solve_linear() fails is the following:


>>> eq1 = a*c*v - a*v + a*v*(1-c)  
>>> solve_linear(eq1,symbols =[a],exclude=[c,v])
(0, 1)

The issue, inside the definition of the solve_linear() method  is with the derivation (dnewn_dxi) of the expression with respect to the symbol(s) we want to solve for. If it is not simplified or at least expanded (an expand_mul() solves the issue for the above examples)  it produces unnecessary free_symbols.

For eq = -a*v - c + (a*b*v - a*v + a – 1)/(b-1) we have :

dnewn_dxi = b*v - v*(b - 1) - v + 1
dnewn_dxi.free_symbols = set([v, b])

Since by simplification/expansion  b*v - v*(b - 1) - v + 1 = 1,  dnewn_dxi.free_symbols  should actually be set([]).